### PR TITLE
feat: install signal handler for graceful shutdown

### LINKED ISSUE #737: Install signal handler for graceful shutdown
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [ ] Question or discussion
- [ ] Bug
- [ ] Requirements
- [x] Feature or performance improvement
- [ ] Other

manager-api doesn't watches on signal channel so the shutdown is forcible, the direct influence is the `Closer` (to cleanup some resources) is useless.

### PR DESCRIPTION
**Pull request by starsz**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance

- Related issues

resolve #737

___
### Requirements or improvement
- Please describe your requirements or improvement suggestions.
Hi, I do some changes to cli_test.sh

1. I think there's no need to change the log file path, so I use `echo > $logfile` to clean up the log file.
2. Create a `check_logfile` function to reduce repetitive code.

### CODE REVIEW

**Comment by nic-chen** [comment]
need test case.
please refer: 
https://github.com/apache/apisix-dashboard/blob/v2.0/api/test/shell/cli_test.sh

**Comment by starsz** [comment]
OK. No problem.

**Comment by juzhiyuan** [comment]
image

Hi, welcome to use the GitHub issue keywords to automatically close the target issue once this PR is merged.

examples:
1. close #9999
2. resolve #9999
3. fix #9999

**Comment by starsz** [comment]
Hi, @nic-chen , I had added a test case. 
Hi, @juzhiyuan, please have a look. Am I doing this right?

**Comment by juzhiyuan** [comment]
Yes! clever you are!

File: api/test/shell/cli_test.sh [hunk 1]
```diff
@@ -34,31 +47,49 @@ go build -o ./manager-api .
...
     echo "failed: should not write info log when level is warn"
     exit 1
 fi
 
-#change level and path
+clean_logfile
+
+#change level
 
-sed -i 's/file_path: logs\/error.log/file_path: .\/error.log/' conf/conf.yaml
```
Comment (nic-chen) [inline api/test/shell/cli_test.sh:49]: please don't delete the existing case，it is for testing log filepath.thanks

**Review by nic-chen — CHANGES_REQUESTED** [review]

File: api/test/shell/cli_test.sh (hunk 1 above)
Comment (starsz) [reply api/test/shell/cli_test.sh:49]: OK.

**Comment by membphis** [comment]
need to rebase your branch @starsz

File: api/test/shell/cli_test.sh [hunk 2]
```diff
@@ -34,29 +47,47 @@ go build -o ./manager-api .
...
+    echo "failed: the manager server didn't deal with signal in correct way"
+    exit 1
+fi
+
+clean_logfile
+
+#change path
 
 sed -i 's/file_path: logs\/error.log/file_path: .\/error.log/' conf/conf.yaml
-sed -i 's/warn/info/' conf/conf.yaml
```
Comment (membphis) [inline api/test/shell/cli_test.sh:50]: why do we need to delete this line? I think it should be useful for old test case.

one PR for one thing, please focus on `install signal handler for graceful shutdown`

**Review by membphis — CHANGES_REQUESTED** [review]

File: api/test/shell/cli_test.sh (hunk 2 above)
Comment (starsz) [reply api/test/shell/cli_test.sh:50]: Hi, I didn't delete this line. This line just is moved to the front.

**Comment by starsz** [comment]
> need to rebase your branch @starsz

OK. Rebased.

**Review by nic-chen — APPROVED** [review]

File: api/test/shell/cli_test.sh [hunk 3]
```diff
@@ -34,29 +47,47 @@ go build -o ./manager-api .
...
     exit 1
 fi
 
-if [[ `grep -c "INFO" ./logs/error.log` -ne '0' ]]; then
-    echo "failed: should not write info log when level is warn"
+clean_logfile
+
+# change level and test signal
+
+sed -i 's/warn/info/' conf/conf.yaml
```
Comment (membphis) [inline api/test/shell/cli_test.sh:61]: I think that is unsafe, should match and replace `level: warn`

File: api/test/shell/cli_test.sh (hunk 2 above)
Comment (membphis) [reply api/test/shell/cli_test.sh:50]: please check your commit log or git operation. 

Your PR cannot contain any other commit that do not belong to you.

**Review by membphis — CHANGES_REQUESTED** [review]

File: api/test/shell/cli_test.sh (hunk 3 above)
Comment (starsz) [reply api/test/shell/cli_test.sh]: That's right. Agree with you.

**Review by membphis — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**